invisible_cities/cities/diomira.py

```diff
@@ -73,30 +82,48 @@ def event_loop(self, NEVT, first_event_no, dataVectors):
         write       = self.writers
         pmtrd       = dataVectors.pmt
         sipmrd      = dataVectors.sipm
-        mc_tracks    = dataVectors.mc
+        mc_tracks   = dataVectors.mc
         events_info = dataVectors.events
+
         for evt in range(NEVT):
+            # Count events in and break if necessary before filtering
+            if self.max_events_reached(self.cnt.counter_value('n_events_tot')):
+                break
+            else:
+                self.cnt.increment_counter('n_events_tot')
+
             # Simulate detector response
-            dataPMT, blrPMT = self.simulate_pmt_response(evt, pmtrd)
-            dataSiPM_noisy = self.simulate_sipm_response(evt, sipmrd, self.noise_sampler)
+            dataPMT, blrPMT = self.simulate_pmt_response(evt, pmtrd,
+                                                         self.sipm_adc_to_pes)
+            dataSiPM_noisy = self.simulate_sipm_response(evt, sipmrd,
+                                                         self.noise_sampler,
+                                                         self.sipm_adc_to_pes)
             dataSiPM = wfm.noise_suppression(dataSiPM_noisy, self.sipms_thresholds)
+            RWF = dataPMT.astype(np.int16)
+            BLR = blrPMT.astype(np.int16)
+
+            # simulate trigger
+            peak_data = self.emulate_trigger(RWF)
```

Comment:
With this new version we are always doing the trigger simulation. Is this what we want? There could be a flag to choose whether to simulate trigger or not. Maybe this is not necessary if we now want to put the trigger always.